### ISSUE TITLE
Build Failed on iOS

### ISSUE THREAD

**Issue by krrevilla** [realm/realm-js#1774]
## Result

Resolved requirements: { SYNC_SERVER_FOLDER: 'sync',
  SYNC_ARCHIVE: 'realm-sync-cocoa-3.0.0.tar.gz',
  SYNC_ARCHIVE_ROOT: 'core' }
No lockfile found at the target, proceeding.
Download url: https://static.realm.io/downloads/sync/realm-sync-cocoa-3.0.0.tar.gz
Extracting realm-sync-cocoa-3.0.0.tar.gz => /Users/pong/Projects/infopais-mobile/node_modules/realm/vendor/realm-ios
{ Error: unexpected end of file
    at Zlib._handle.onerror (zlib.js:371:17) errno: -5, code: 'Z_BUF_ERROR' }
Command /bin/sh failed with exit code 1

## Steps to Reproduce

react-native init Test
yarn add realm
react-native link realm
react-native run-ios

## Version of Realm and Tooling

- Realm JS SDK Version: 2.3.0
- Node or React Native: React Native - 0.55
- Client OS & Version: macOS High Sierra
- Which debugger for React Native: ?/None

Issue is same with #1647, but after removing and installing after many times. Still getting error

**Comment by krrevilla** [comment]
Issue is same with [#1647](https://github.com/realm/realm-js/issues/1647), but after removing and installing after many times. Still getting error

**Comment by kneth** [comment]
It looks like downloaded `realm-sync-cocoa-3.0.0.tar.gz` failed silently, and it is not possible to unpack it. I see it sometimes on our CI system, and typically I just retry. We are using Amazon S3 to host our files, and the service is apparently not 100 % reliable.

**Comment by krrevilla** [comment]
Okay got it @kneth will try and try for now. Thanks

**Comment by rcalabro** [comment]
I am getting the same error with realm-sync-cocoa-2.2.9.tar.gz
I tried to clean caches and reinstall several times now
Any ideia how to solve this ?

**Comment by kneth** [comment]
@rcalabro Sync 2.2.9 is from December 2017, and it is probably not work with `master` of this repository. Realm JS depends heavily on Realm Sync and Realm Core so many changes to those two will be reflected in the code base. I suggest that you try to build directly from `master` with recent versions of sync and core.

**Comment by ekeuus** [comment]
Tried it multiple times with various versions as well, can't get it to work. Debugging shows me that

const corePath = desired.CORE_ARCHIVE && path.resolve(os.tmpdir(), desired.CORE_ARCHIVE);

is undefined

**Comment by rcalabro** [comment]
Updating Xcode was the only "solution" 






Get Outlook for Android







On Fri, May 18, 2018 at 6:31 PM -0300, "Ekke Tõiv Uustalu" <notifications@github.com> wrote:












Tried it multiple times with various versions as well, can't get it to work. Debugging shows me that


const corePath = desired.CORE_ARCHIVE && path.resolve(os.tmpdir(), desired.CORE_ARCHIVE);


is undefined

**Comment by ekeuus** [comment]
Lowering the version to 2.3.1 and clearing yarn caches worked for me.

**Comment by krrevilla** [comment]
@kneth no, i don't experience it anymore. Somehow internet connection can also cause this issue.

**Comment by kneth** [comment]
@krrevilla Thanks. I close the issue. Never hesitate to ask another time.

**Comment by dwilt** [comment]
This happened to us. As @ekeuus suggested, downgrading to 2.3.1 solved it.

**Comment by kneth** [comment]
@dwilt Which version did you try before downgrading?

**Comment by dwilt** [comment]
![image](https://user-images.githubusercontent.com/1065539/42382383-1f231da6-8102-11e8-8194-148d941add47.png)
![image](https://user-images.githubusercontent.com/1065539/42382428-4166b3aa-8102-11e8-92ef-d3c126f236e9.png)
@kneth It's happening to me with the latest version (2.12.0)

**Comment by kneth** [comment]
@dwilt I have tried to manually download the file, and it seems to be fine:

    $ curl -O https://static.realm.io/downloads/sync/realm-sync-cocoa-3.7.0.tar.gz
    $ tar tzf realm-sync-cocoa-3.7.0.tar.gz

Can you try the two command above to see if you get any error messages?

**Comment by dwilt** [comment]
@kneth Maybe we're onto something here. Just tried the `curl`:

![image](https://user-images.githubusercontent.com/1065539/42450580-a2ce17bc-8352-11e8-878b-a16e1478ff1c.png)

To make sure it wasn't my internet:

![image](https://user-images.githubusercontent.com/1065539/42450599-b55e1594-8352-11e8-9a9b-d22272f6dbda.png)

Any ideas?

**Comment by kneth** [comment]
* Do you have a proxy server in front?
* Does it make a difference if you add `--http2` or `--http1.1` to the curl command?

**Comment by dwilt** [comment]
@kneth I'm pretty sure I don't have a proxy server. I'd know that, right? You can't just have one of those without realizing it? 

Both didn't work:
![image](https://user-images.githubusercontent.com/1065539/42454239-f8239fb6-835c-11e8-8654-0b9fe2e1f959.png)

**Comment by kneth** [comment]
Yes, you will know if you are behind a proxy as it will require some configuration.

This is a bit bizarre. I mean, it works perfectly for me - and I am in Europe and our download server is in USA.

**Comment by dwilt** [comment]
Is this what is happening during the build process? It's trying to
redownload this tar file?

**Comment by kneth** [comment]
It will download the file the first time.

Btw, does `wget https://static.realm.io/downloads/sync/realm-sync-cocoa-3.7.0.tar.gz` work?

**Comment by dwilt** [comment]
@kneth Had to download `wget` via `brew` but it worked:

![image](https://user-images.githubusercontent.com/1065539/42468522-25e2062e-8383-11e8-97f6-a3a8e470c554.png)
![image](https://user-images.githubusercontent.com/1065539/42468548-372e2c82-8383-11e8-9f64-b2b27c301f32.png)

**Comment by kneth** [comment]
I guess we can conclude that the problem is the download script (`scripts/download-realm.js`) and not our server. The script is using [`node-fetch`](https://www.npmjs.com/package/node-fetch), and `curl` has the same issue.

**Comment by dwilt** [comment]
What does that mean? Is there a way we can work around this in the mean time?

**Comment by kneth** [comment]
You can unpack the file in `vendor/realm-mac-x64` and create an empty file `download-realm.lock` (`touch vendor/realm-mac-x64/download-realm.lock`).

**Comment by kneth** [comment]
And then I have to understand in which situations, `node-fetch` doesn't work 😄

**Comment by dwilt** [comment]
@kneth where is this `vendor` folder at?

**Comment by dwilt** [comment]
![image](https://user-images.githubusercontent.com/1065539/42472674-2de6307c-8390-11e8-930a-7c1df71abee7.png)

Are you talking about here: `node_modules/realm/vendor`?

**Comment by kneth** [comment]
Yes, the one in your screenshot

**Comment by dwilt** [comment]
@kneth I tried that and it still failed to build.

**Comment by dwilt** [comment]
@kneth tried downgrading Xcode, moving over to the same node version as another guy on my team, and still nothing. I even tried creating a brand new react native project and just installed realm - same build error.

**Comment by kneth** [comment]
Maybe https://realm.io/docs/javascript/latest/#cannot-download-realm-sync-cocoa can help you.

**Comment by dwilt** [comment]
@kneth those instructions aren't working for me either. I noticed that the latest package is 3.8.0 so I updated the script. I tried putting the files in the `realm-ios` folder but they just get wiped every time a build runs. Isn't the `download-realm.lock` file supposed to prevent it from trying to sync from the server?

What I find really interesting, is that if I'm offline, it still tries to sync and then overwrites my folder. Our project is completely blocked by this. Any help (happy to do a screenshare here) would be great appreciated.

**Comment by dwilt** [comment]
I created a video showing my steps. Let me know if something seems off here.

[![realm trouble](https://img.youtube.com/vi/9Mq3nrcOwJ0/0.jpg)](https://www.youtube.com/watch?v=9Mq3nrcOwJ0)

**Comment by dwilt** [comment]
Also, fwiw, I can `curl` now but the build still fails.

**Comment by dwilt** [comment]
Reformatted my Mac. All is working now.

**Comment by joaom182** [comment]
@dwilt you reformatted your MAC? :O WTF!!

**Comment by joaom182** [comment]
Not able to download

`https://static.realm.io/downloads/sync/realm-sync-cocoa-3.8.0.tar.gz Failed network error`

Anyone can post this file in another source to test?

**Comment by joaom182** [comment]
My issue is related with Wifi on OSX High Sierra. Trying to download with Windows PC, it's fine.

**Comment by dwilt** [comment]
@joaom182 haha yea, I tried everything else. Reformatting was the last option.

**Comment by snokier** [comment]
Oh my god! I have the same issue. I really can't imagine Reformatting is the only way!

**Comment by joaom182** [comment]
@snokier I did not have to format, i have only to fix the issue with my Wi-fi on High Sierra.

**Comment by kneth** [comment]
I have been offline for a while (vacation), and it is interesting to read. MacOS has been having issues with WiFi but I thought it was over. @joaom182 Can you explain (I am just curious here) what your WiFi issue was?

@dwilt I admire that you go the extra mile (formatting your Mac) just to use Realm - thanks!

**Comment by joaom182** [comment]
@kneth Instability, many packet loses, so the download never complete.

This resolved my issue:
http://osxdaily.com/2016/09/22/fix-wi-fi-problems-macos-sierra/

**Comment by alwex** [comment]
I know this issue has been closed, but I just had the issue recently and found what was causing this.
Here is what happens to me, a corrupted file have been downloaded on the build server during a previous build, or a savagely aborted build, I can't really tell. But I was having the exact same error message as described above.
To find out what was going on, I decided to start logging every steps on the `download-realm.js` file and found out that the download is not triggered if a file is already present on a temporary folder, was this one for me `/var/folders/5v/lgs_9f8d6ms25pf_09y2xvfc0000gn/T/realm-sync-cocoa-2.2.10.tar.gz`

Fixing the issue was then just about deleting this corrupted file and running the build again.
This would explain the issue being resolved after a restart or even after a format as well as the issue encountered by some when on an unstable network.

Hope this can help someone.
